### PR TITLE
🛂 Auth service deployment

### PR DESCRIPTION
**Pull request by JoelSaleem**
Changes:
1. Change dockerfile for our use case (multi-stage build and tls handled by traefik)
2. helm values files for dev and prod
3. circleci
4. helm chart moved out of this repo (will live in charts)

The last change means it could be difficult to update if we want to merge changes from the master project, however, I think it could be more confusing if we have some helm charts living in the charts repo, and some in the application code repos.

### CODE REVIEW

**Review by sergiurqv — APPROVED** [review]

**Comment by sergiurqv** [comment]
Lovely commit messages

File: config/config.go [hunk 1]
```diff
@@ -112,7 +136,19 @@ func (c *Config) RunServer() (chan error, net.Listener) {
 		mux.Handle("/metrics", promhttp.HandlerFor(registry, promhttp.HandlerOpts{}))
 		mux.Handle("/ping", pingHandler)
 		mux.Handle("/", histogramMw(loggingMiddleWare(handler)))
-		srv.Handler = mux
+
+		h := cors.New(cors.Options{
+			Debug:          true,
+			AllowedOrigins: []string{"*"},
```
Comment (udoniyor) [inline config/config.go:142]: similar to the comment in the middleware, would scope this differently based on the env

File: config/config.go [hunk 2]
```diff
@@ -112,7 +136,19 @@ func (c *Config) RunServer() (chan error, net.Listener) {
 		mux.Handle("/metrics", promhttp.HandlerFor(registry, promhttp.HandlerOpts{}))
 		mux.Handle("/ping", pingHandler)
 		mux.Handle("/", histogramMw(loggingMiddleWare(handler)))
-		srv.Handler = mux
+
+		h := cors.New(cors.Options{
+			Debug:          true,
```
Comment (udoniyor) [inline config/config.go:141]: Do we want debug true in prod?

File: config/config.go [hunk 3]
```diff
@@ -90,6 +91,29 @@ func (c *Config) PingHandler(rw http.ResponseWriter, r *http.Request) {
 	return
 }
 
+// type MyServer struct {
```
Comment (udoniyor) [inline config/config.go:94]: Is the commented code needed?

Comment (JoelSaleem) [reply config/config.go:94]: Nope, my bad

File: config/config.go (hunk 2 above)
Comment (JoelSaleem) [reply config/config.go:141]: nope. will, remove

**Review by udoniyor — APPROVED** [review]

**Pull request merged**